projects/microcontrollers/src/main/kotlin/microcontrollers/WirelessLinkMicrocontroller.kt

```diff
@@ -15,7 +15,11 @@ class WirelessLinkMicrocontroller(override val lock: Lock, override val recv: ()
     }
 
     fun receive(): WirelessLinkReceiveMessage {
-        return WirelessLinkReceiveMessage(writeMessage(Message(0x03, byteArrayOf(0x00))))
+        val msg = writeMessage(Message(0x03, byteArrayOf(0x00)))
+        return when (msg.command.toInt()) {
+            0x03 -> WirelessLinkReceiveMessage(msg)
+            else -> WirelessLinkReceiveMessage(Message(0x03, ByteArray(payloadSizes[msg.command]!!, { 0 })))
```

Comment:
`payloadSizes[msg.command]!!` is reusing the error command from the response. It should instead be `0x03`.